server/service/queries.go

```diff
@@ -287,17 +291,32 @@ func (svc *Service) ModifyQuery(ctx context.Context, id uint, p fleet.QueryPaylo
 	}
 	if p.Logging != nil {
 		query.Logging = *p.Logging
+		if *p.Logging != fleet.LoggingSnapshot {
+			shouldDiscardQueryResults = true
+		}
 	}
 	if p.ObserverCanRun != nil {
 		query.ObserverCanRun = *p.ObserverCanRun
 	}
+	if p.DiscardData != nil {
+		query.DiscardData = *p.DiscardData
+		if *p.DiscardData {
+			shouldDiscardQueryResults = true
+		}
+	}
 
 	logging.WithExtras(ctx, "name", query.Name, "sql", query.Query)
 
 	if err := svc.ds.SaveQuery(ctx, query); err != nil {
 		return nil, err
 	}
 
+	if shouldDiscardQueryResults {
+		if err := svc.ds.DeleteAllResultsForQuery(ctx, id); err != nil {
```

Comment:
Should we do similarly to ApplyQueries? Basically add a `deleteResults bool` to `SaveQuery` and execute all in a transaction.

Reply:
I see, yeah can add that as a transaction.